File: leetcode/201. Bitwise AND of Numbers Range/201._Bitwise_AND_of_Numbers_Range_1.py
class Solution(object):
    def rangeBitwiseAnd(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        diff = n-m
        tailZeros = 0
        while diff > 0:
            diff = diff >> 1
            tailZeros += 1
        mask = (~0) << tailZeros
        return m & n & mask

File: leetcode/201. Bitwise AND of Numbers Range/201._Bitwise_AND_of_Numbers_Range_1.java
public class Solution {
    public int rangeBitwiseAnd(int m, int n) {
        int temp = log2(n-m);
        int tailMask = (0xffffffff << temp);
        return (m & n & tailMask);
    }
    public int log2(int k) {
        int ans = 0;
        while( k > 0) {
            k = k >> 1;
            ans++;
        }
        return ans;
    }
}